modules/auxiliary/scanner/http/http_login.rb

```diff
@@ -44,6 +44,12 @@ def initialize
       ])
     register_autofilter_ports([ 80, 443, 8080, 8081, 8000, 8008, 8443, 8444, 8880, 8888 ])
 
+    register_advanced_options(
+      [
+        OptBool.new('RedirectsOnSuccess', [ false, 'Login may redirect on successful login', false]),
+      ]
+    )
```

Comment:
I'm worried this won't provide additional extension points in the future. It's maybe a small bit of scope creep, but I wonder if we could use a similar pattern to other login bruteforcers, and allow user configurable port range, i.e. `200,201,300-399`, as well as the ability to regex match response text for success/failure scenarios

Reply:
I not sure accepting a `port` list for status codes, that seems like shoehorning function into the pattern the class is not really meant for and validation would need to be stricter. I could see an advanced option for `LoginHttpSuccessCodes` however that feels like a new `Opt::HttpCodes` type do we really want to expand that much?

As a generic `http` login utility I think a regex match might bit *too* flexible, I like the idea of adding a flexible response `content` attribute or something. It seems like that could get dicey to maintain as each application can do this differently via `headers` or `body` content and the may also introduce the idea of needing to support more complex combinations, like `302` response code and content or something. Scenarios of that sort are a reason to create a new unique login scanner class in the first place.

Reply:
Adjusted to take `OptString` and parse values into an array of integers for validation done during processing of the first host in the targeted and failing the module early in the process if not valid.

The login scanner can now in theory raise an `ArgumentError` that might deserve a `rescue` but I deferred that for now.